openhexa/sdk/pipelines/runtime.py

```diff
@@ -1,22 +1,162 @@
+import ast
 import base64
-import importlib
+import dataclasses
+import enum
 import io
+import importlib
+
 import os
+import requests
 import sys
+import typing
+
 from zipfile import ZipFile
+from pathlib import Path
+from .pipeline import Pipeline
 
-import requests
 
-from .pipeline import Pipeline
+class PipelineNotFound(Exception):
+    pass
+
+
+@dataclasses.dataclass
+class PipelineParameterSpecs:
+    code: str
+    type: typing.Union[typing.Type[str], typing.Type[int], typing.Type[bool]]
+    name: typing.Optional[str] = None
+    choices: typing.Optional[typing.Sequence] = None
+    help: typing.Optional[str] = None
+    default: typing.Optional[typing.Any] = None
+    required: bool = True
+    multiple: bool = False
+
+
+@dataclasses.dataclass
+class PipelineSpecs:
+    code: str
+    name: str
+    parameters: typing.Sequence[PipelineParameterSpecs] = dataclasses.field(default_factory=list)
+    timeout: int = None
+
+
+class ImportStrategy(enum.Enum):
+    AST = "AST"
+    IMPORT = "IMPORT"
+
+
+def import_pipeline(pipeline_dir_path: str, strategy: ImportStrategy):
```

Comment:
See remark above. This function should not be called from outside this module, and should not deal with a strategy.